app/assets/javascripts/media_player_wrapper/avalon_player_new.es6

```diff
@@ -217,70 +220,92 @@ class MEJSPlayer {
         .data('mediaObjectId');
 
       // Update helper object noting we want the new media clip to auto start
-      this.switchPlayerHelper = {
+      t.switchPlayerHelper = {
         active: true,
         data: {},
         paused: false
       };
 
-      // Get the new stream info
-      $('.media-show-page').removeClass('ready-to-play');
-        $.ajax({
-          url: `/media_objects/${mediaObjectId}/section/${sectionId}/stream`,
-          dataType: 'json',
-          data: {
-            content: sectionId
-          }
-        })
-          .done(response => {
-            this.setContextVars(response);
-            // Switch media in the current player instance
-            this.switchMedia(mediaElement, originalNode, instance, response);
-            this.updateShareLinks();
-          })
-          .fail(error => {
-            console.log('error', error);
-          });
+       // Go to next section
+       t.getNewStreamAjax(
+        `/media_objects/${mediaObjectId}/section/${sectionId}`,
+        true
+      );
     }
   }
 
   /**
    * Swap the media source and track in the current player instance with values from
    * the new stream info
-   * @param media {Object} - MediaElement wrapper
-   * @param node {Object} - HTML node
-   * @param instance {Object} - player instance
+   * @function switchPlayer
+   * @param {Boolean} isEnded
+   * @param {Object} playlistItemsT
+   * @returns {void}
    */
-  switchMedia(media, node, instance) {
+  switchPlayer(isEnded, playlistItemsT) {
     let markup = '';
 
-    node.innerHTML = '';
+    this.node.innerHTML = '';
     this.currentStreamInfo.stream_hls.map(source => {
       markup += `<source src="${
         source.url
       }" type="application/x-mpegURL" data-quality="${source.quality}"/>`;
     });
 
-    // Add captions
     if (this.currentStreamInfo.captions_path) {
       markup += `<track srclang="en" kind="subtitles" type="${
         this.currentStreamInfo.captions_format
       }" src="${this.currentStreamInfo.captions_path}"></track>`;
     }
 
-    node.innerHTML = markup;
+    this.node.innerHTML = markup;
+
+    // Handle 'canplay' events fired by player
+    this.mediaElement.addEventListener(
+      'canplay',
+      this.handleCanPlay.bind(this)
+    );
+
+    this.player.buildquality(this.player, null, null, this.mediaElement);
+    this.player.buildtracks(this.player, null, this.player.layers, this.mediaElement);
 
-    node.player.buildquality(instance, null, null, media);
-    node.player.buildtracks(instance, null, instance.layers, media);
+    // Build playlists button from the new stream when not in playlists
+    if(!playlistItemsT) {
+      this.player.options.playlistItemDefaultTitle = this.currentStreamInfo.embed_title;
+      this.player.buildaddToPlaylist(this.player, null, null, null);
+    }
 
+    // Turn on captions
     this.toggleCaptions();
 
-    instance.load();
-    instance.play();
+    this.player.load();
+    if(isEnded) {
```

Comment:
Now that I think of it, we don't need this conditional check at all because it is already checked in `getNewStreamAjax` function before calling `switchPlayer`. So `isEnded` is always true.

`this.switchPlayerHelper.paused` is initialized to `false` and changed with the status of the player within `avalon_player_new.es6`. So when using `this.switchPlayerHelper.paused` when invoking this function from an external file, the value of it is always `false`.